cli/src/config.rs

```diff
@@ -268,62 +257,84 @@ impl ConfigEnv {
         }
     }
 
-    fn config_path(self) -> Result<ConfigPath, ConfigError> {
+    // Returns a config path, potentially creating a new one if told. Can return
+    // None if no config exists and we're not creating a new one.
+    // Also bubbles up errors if encountered.
+    fn new_or_existing_config_path(
+        self,
+        handle_missing: HandleMissingConfig,
+    ) -> Result<Option<PathBuf>, ConfigError> {
         if let Some(path) = self.jj_config {
             // TODO: We should probably support colon-separated (std::env::split_paths)
-            return Ok(ConfigPath::new(Some(PathBuf::from(path))));
+            let path = PathBuf::from(path);
+            if handle_missing == HandleMissingConfig::Create && !path.exists() {
+                create_config_file(&path)?;
+            }
+
+            return Ok(Some(path));
         }
-        // TODO: Should we drop the final `/config.toml` and read all files in the
-        // directory?
-        let platform_config_path = ConfigPath::new(self.config_dir.map(|mut config_dir| {
+        // look for existing configs first
+        let platform_config = self.config_dir.as_ref().map(|config_dir| {
+            let mut config_dir = config_dir.clone();
             config_dir.push("jj");
             config_dir.push("config.toml");
             config_dir
-        }));
-        let home_config_path = ConfigPath::new(self.home_dir.map(|mut home_dir| {
-            home_dir.push(".jjconfig.toml");
-            home_dir
-        }));
-        use ConfigPath::*;
-        match (platform_config_path, home_config_path) {
-            (Existing(platform_config_path), Existing(home_config_path)) => Err(
-                ConfigError::AmbiguousSource(platform_config_path, home_config_path),
-            ),
-            (Existing(path), _) | (_, Existing(path)) => Ok(Existing(path)),
```

Comment:
In the commit message, you write this:

> * if told to create missing, it creates one at the platform config location (not $HOME)

It looks to me like this line made it so the old behavior already was that. Am I missing something or is the commit message incorrect (maybe stale)?

Reply:
You're right, I'll update the description.